examples/tfengine/app.hcl

```diff
@@ -20,46 +20,267 @@ template "app" {
   data = {
     constants = {
       shared = {
-        env_code        = "s"
-        folder_id       = "123"
-        billing_account = "000"
-        project_prefix  = "example-prefix"
-        state_bucket    = "example-state"
-        region          = "us-central1"
+        env_code            = "s"
+        folder_id           = "123"
+        billing_account     = "000"
+        project_prefix      = "example-prefix"
+        state_bucket        = "example-state"
+        bigquery_location   = "us-east1" # No BigQuery region in us-central1.
+        cloud_sql_region    = "us-central1"
+        cloud_sql_zone      = "a"
+        compute_region      = "us-central1"
+        gke_region          = "us-central1"
+        storage_location    = "us-central1"
+        secret_locations    = ["us-central1"]
+        healthcare_location = "us-central1"
       }
       dev = {
         env_code  = "d"
         folder_id = "456"
       }
     }
     devops = {
-      admins_group  = "admins@example.com"
-      devops_owners = ["devops-owners-group@example.com"]
+      admins_group       = "admins@example.com"
+      devops_owners      = ["group:devops-owners-group@example.com"]
+      enable_gcs_backend = false # TODO(user): change to true after initial devops deployment.
     }
     deployments = [
+      {
+        name = "project_secrets",
+        resources = {
+          project = {
+            name_suffix = "secrets"
+            apis = ["secretmanager.googleapis.com"]
+          }
+          secrets = [
+            {
+              secret_id = "manual-sql-db-user"
+            },
+            {
+              secret_id   = "auto-sql-db-password"
+              secret_data = "$${random_password.db.result}" // Use $$ to escape reference.
+            },
+          ]
+        }
+        terraform_addons = {
+      raw_config = <<EOF
+resource "random_password" "db" {
+  length = 16
+  special = true
+}
+EOF
+        }
+      },
       {
         name = "project_networks",
         resources = {
           project = {
```

Comment:
I like this. Treating the project just like a resource, and we can also make output of this project resource available to other resources.